### PR TITLE
[refactor] detection script

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- refactored detection script
- add ability to save also as xml
- add minor test

Any feedback is welcome :hugs:

### CODE REVIEW

File: scripts/detect_text.py [hunk 1]
```diff
@@ -24,95 +24,71 @@
...
+    if out_format == "json":
+        output = json.dumps(out.export(), indent=2)
+    elif out_format == "txt":
+        output = out.render()
+    elif out_format == "xml":
+        output = out.export_as_xml()
+
+    path = Path("output").joinpath(file_path.stem + "." + out_format)
+    if out_format == "xml":
+        for i, (xml_bytes, xml_tree) in enumerate(output):
```
Comment (ianardee) [inline scripts/detect_text.py:54]: getting a warning here about an unused variable

File: scripts/detect_text.py [hunk 2]
```diff
@@ -24,95 +24,71 @@
...
 
 
 if __name__ == "__main__":
     parsed_args = parse_args()
-    try:
-        main(parsed_args)
-    except KeyboardInterrupt:
-        print("Cancelled")
-        pass
+    main(parsed_args)
```
Comment (ianardee) [inline scripts/detect_text.py:94]: why take out the keyboard interrupt?

**Review by ianardee — APPROVED** [review]

File: scripts/detect_text.py (hunk 1 above)
Comment (felixdittrich92) [reply scripts/detect_text.py:54]: Hi 👋 yeah we could replace it with _ but so it is more clear what it returns :)

File: scripts/detect_text.py (hunk 2 above)
Comment (ianardee) [reply scripts/detect_text.py:94]: yes ok without

File: scripts/detect_text.py (hunk 1 above)
Comment (felixdittrich92) [reply scripts/detect_text.py:54]: The XML export returns a list which contains a tuple (XML as bytes, the whole ElementTree) for each page :)

Comment (ianardee) [reply scripts/detect_text.py:54]: ok clear

thanks a lot for the cleanup work on the PR :-)

Comment (felixdittrich92) [reply scripts/detect_text.py:54]: No problem 🤗

**Pull request merged**